### PR TITLE
Rhi implementation

### PR DESCRIPTION
**Pull request by blakdragan7**
Initital merge into development for tech preview

### CODE REVIEW

File: Source/Rive/Private/Rive/RiveTexture.cpp [hunk 1]
```diff
@@ -12,7 +12,7 @@
 
 URiveTexture::URiveTexture()
 {
-    SRGB = true;
+    SRGB = false;
```
Comment (schetle) [inline Source/Rive/Private/Rive/RiveTexture.cpp:15]: Has this change been tested on other platforms (e.g.: macOS)? Previously, macOS required specific sRGB settings (for example, the inclusion of the flag in .SetFlags below, but I'm not sure if this specific boolean in the texture will have any affect.

File: Source/Rive/Private/RiveModule.cpp [hunk 2]
```diff
@@ -18,11 +18,16 @@ THIRD_PARTY_INCLUDES_END
 
 void FRiveModule::StartupModule()
 {
+    // for some reason the shader file path is not loading correctly. 
+    FString PluginShaderDir = FPaths::Combine(IPluginManager::Get().FindPlugin(TEXT("Rive"))->GetBaseDir(), TEXT("Shaders"));
+    AddShaderSourceDirectoryMapping(TEXT("/Plugin/Rive"), PluginShaderDir);
```
Comment (schetle) [inline Source/Rive/Private/RiveModule.cpp:23]: Should this be TEXT("/Plugins/Rive") (plural Plugins)

File: Source/RiveRenderer/Private/Platform/RiveRendererD3D11.h [hunk 3]
```diff
@@ -11,11 +11,18 @@ THIRD_PARTY_INCLUDES_START
 #include "Windows/AllowWindowsPlatformTypes.h"
 #include "Windows/MinWindows.h"
 #include "rive/renderer/d3d/render_context_d3d_impl.hpp" // This ends up including windows.h 
+#include "rive/renderer/d3d/render_context_d3d_impl.hpp" // This ends up including windows.h
```
Comment (schetle) [inline Source/RiveRenderer/Private/Platform/RiveRendererD3D11.h:14]: Duplicate line

File: Source/RiveRenderer/Private/Platform/pls_render_context_rhi_impl.hpp [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+#include "rive/renderer/rive_renderer.hpp"
+#include "rive/renderer/rive_render_image.hpp"
+#include "rive/renderer/buffer_ring.hpp"
+THIRD_PARTY_INCLUDES_END
+
+namespace rive
+{
+namespace gpu
+{
+class RIVERENDERER_API RenderTargetRHI : public RenderTarget
```
Comment (schetle) [inline Source/RiveRenderer/Private/Platform/pls_render_context_rhi_impl.hpp:17]: Unreal is unfriendly to classes nested inside of namespaces (you won't see Unreal-proper classes inside of namespaces). I suggest not embedding Unreal classes inside of namespaces.

**Review by schetle — COMMENTED** [review]
A few comments to review

File: Source/Rive/Private/RiveModule.cpp (hunk 2 above)
Comment (blakdragan7) [reply Source/Rive/Private/RiveModule.cpp:23]: I don't think so since its working.

File: Source/Rive/Private/Rive/RiveTexture.cpp (hunk 1 above)
Comment (blakdragan7) [reply Source/Rive/Private/Rive/RiveTexture.cpp:15]: I have not tested in any environment other then my laptop. I will remove this for now just in case.

**Pull request merged**